raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -541,9 +539,13 @@ contract TokenNetwork is Utils {
 
         emit ChannelUnlocked(channel_identifier, participant, unlocked_amount, returned_tokens);
 
-        assert(computed_locksroot != 0);
-        assert(unlocked_amount > 0);
-        assert(locked_amount > 0);
+        // Channel must be settled and channel data deleted
+        require(channels[channel_identifier].state == 0);
+
+        require(computed_locksroot != 0);
+        require(locked_amount > 0);
+        assert(locked_amount >= unlocked_amount);
+        assert(locked_amount >= returned_tokens);
```

Comment:
If this is an overflow check for [this](https://github.com/raiden-network/raiden-contracts/pull/64/files#diff-2f6fb78ac7aaf29aaec1ed8ab59468f8R522) line: `returned_tokens = locked_amount - unlocked_amount;`, then with the logic explained above it should still be a `require()`